doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,235 @@
+after the first few "obvious" ones should *require* the use of the keyword
+when given.  This is implemented by including ``*`` at the appropriate point
+in the signature.
+
+For example, a function ``foo`` that operates on a single array but that has
+several optional parameters (say ``method``, ``flag``, ``rtol`` and ``atol``)
+would be defined as::
+
+    def foo(x, *, method='basic', flag=False, rtol=1.5e-8, atol=1-12):
+        ...
+
+To call ``foo``, all parameters other than ``x`` must be given with an
+explicit keyword, e.g. ``foo(arr, rtol=1e-12, method='better')``.
+
+This forces callers to give explicit keyword parameters (which most users
+would probably do anyway even without the use of ``*``), *and* it means
+additional parameters can be added to the function anywhere after the
+``*``; new parameters do not have to be added after the existing parameters.
+
+
+Test functions from `numpy.testing`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In new code, don't use `assert_almost_equal`, `assert_approx_equal` or
+`assert_array_almost_equal`. This is from the docstrings of these
+functions::
+
+    It is recommended to use one of `assert_allclose`,
+    `assert_array_almost_equal_nulp` or `assert_array_max_ulp`
+    instead of this function for more consistent floating point
+    comparisons.
+
+For more information about writing unit tests, see `Testing Guidelines
+<https://docs.scipy.org/doc/numpy/reference/testing.html>`_.
+
+Testing that expected exceptions are raised
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When writing a new test that a function call raises a certain exception,
+the preferred style is to use ``pytest.raises`` as a context manager, with
+the code that is supposed to raise the exception in the code block defined
+by the context manager.  The `match` keyword argument of ``pytest.raises``
+is given with enough of the expected error message attached to the exception
+to ensure that the expected exception is raised.
+
+For example, the function `scipy.stats.zmap` is supposed to raise a
+``ValueError`` if the input contains ``nan`` and ``nan_policy`` is ``"raise"``.
+A test for this is::
+
+    scores = np.array([1, 2, 3])
+    compare = np.array([-8, -3, 2, 7, 12, np.nan])
+    with pytest.raises(ValueError, match='input contains nan'):
+        stats.zmap(scores, compare, nan_policy='raise')
+
+The ``match`` argument ensures that the test doesn't pass by raising
+a ``ValueError`` that is not related to the input containing ``nan``.
+
+
+A parameter called lambda
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are functions where convention dictates that (ideally) a parameter
+should be called 'lambda'.  In Python, 'lambda' is a reserved word that
+cannot be used as a variable name.  If a new function or class has a
+parameter where a name that is similar to 'lambda' is really the best
+choice, the preferred name to use is [TBD: ``lam``, ``lmb``, ``lamb``,
+``lmbda``, ``lambda_``?].
+
+There has been a lack of consistency in the past:
```

Comment:
Nice.  I'm not sure it would be worth the churn, however.  In fact, I'm removing this section from the PR.  If it turns out that there is interest in having an explicit guideline for the name to use for a parameter that is naturally called `lambda` (or perhaps more general guidelines for choosing parameter names), we can handle it in a follow-up PR.